### ISSUE TITLE
Validate UCUM unit strings

### ISSUE THREAD

**Issue by lmsurpre** [IBM/FHIR#694]
The valueset is defined at https://www.hl7.org/fhir/valueset-ucum-units.html

This has a required binding in these three resources:
```
Resource: ResearchElementDefinition.characteristic.unitOfMeasure (CodeableConcept / Required)
Resource: EffectEvidenceSynthesis.effectEstimate.unitOfMeasure (CodeableConcept / Required)
Resource: RiskEvidenceSynthesis.riskEstimate.unitOfMeasure (CodeableConcept / Required)
```

Per #601, we should validate that those values are valid when we validate those resource types.

Consider using https://github.com/FHIR/Ucum-java (or https://unitsofmeasurement.github.io)

**Comment by lmsurpre** [comment]
note:  https://github.com/FHIR/Ucum-java  seems to be limited to just the common set of UCUM codes.
For a more complete implementation we could use "Indria" (JSR something), but its probably not needed.

**Comment by lmsurpre** [comment]
note: we have some UCUM length unit support for our near search.  After we have improved UCUM support we should consider consolidating that code with whatever we introduce.

**Comment by JohnTimm** [comment]
Using Indriya (the RI for JSR-385), basic format syntax validation can be achieved with something like the following:
```
SystemOfUnitsService service = ServiceProvider.current();
UnitFormat format = service.getFormatService().getUnitFormat();
Unit<?> unit = format.parse(unit)
```
`UnitFormat.parse` will throw a runtime exception (`MeasurementParseException`) if the syntax is not valid.

**Comment by michaelwschroeder** [comment]
Validation of codes for the ucum-units value set is being done by the UCUM parser added under PR #1420. In addition to that value set, the http://hl7.org/fhir/ValueSet/all-distance-units and http://hl7.org/fhir/ValueSet/all-time-units value sets will also have to be validated syntactically. They are referenced as maxValueSets in the Age, Distance, and Duration types. That work will be done against this issue in a subsequent PR.

**Comment by lmsurpre** [comment]
I submitted a RiskEvidenceSynthesis resource with a bogus ucum code, but the server said it was valid:
```{
  "resourceType": "RiskEvidenceSynthesis",
  "id": "example",
  "text": {
    "status": "generated",
    "div": "<div xmlns=\"http://www.w3.org/1999/xhtml\">[Put rendering here]</div>"
  },
  "status": "draft",
  "population": {
    "reference": "EvidenceVariable/example"
  },
  "outcome": {
    "reference": "EvidenceVariable/example"
  },
  "riskEstimate": {
    "unitOfMeasure": {
      "text": "F",
      "coding": [{
        "system": "http://unitsofmeasure.org",
        "code": "lkjlkajsdflkjz"
      }]
    }
  }
}```

I think we need to be more stringent for this UCUM validation to actually be useful.

**Comment by lmsurpre** [comment]
Discussed with the team and, since the description in this issue was vague abut what it means for a UCUM code to be "valid", we decided to consider this one as complete. I opened #1444 for future improvements, but that is currently blocked by #1418.

Additionally, we agreed that #1216 should be prioritized so that we get warnings when the units for Age, Distance, and Duration unit codes are not in the Extensible valuesets for those issues.
